pkg/manifests/manifests.go

```diff
@@ -1513,6 +1513,11 @@ func (f *Factory) setupGoGC(p *monv1.Prometheus) {
 		// be savvy on CPU hence set GOGC=100 (Go runtime default) in this
 		// case.
 		p.Spec.Containers[i].Env = append(p.Spec.Containers[i].Env, v1.EnvVar{Name: "GOGC", Value: "100"})
+
+		// Until we're certain setting GOMEMLIMIT to 0.9 (default ratio) of detected maximum
+		// container or system memory won't result in excessive CPU usage, we're disabling the
+		// auto setting for SNO.
+		p.Spec.Containers[i].Args = append(p.Spec.Containers[i].Args, "--no-auto-gomemlimit")
```

Comment:
should rather use `p.Spec.AdditionalArgs`?

Reply:
Good catch, even though we're overriding all the args, no test seems to be failing...
e.g. https://github.com/openshift/cluster-monitoring-operator/blob/5ee3bd44ad0dd03b3e55a8693d9b0918aa613953/test/e2e/config_test.go#L231
I'll take a look
/hold

Reply:
I'll check with the SNO team, we'd probably pick the same upgrade one than payload...